uwblueprint/jump-math
# feat: Return StudentDTOs in testSession Resolver

## Notion ticket link

[Return StudentDTO within testSession Resolver](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)


## Implementation description
* Map student results to StudentDTOs in `testSession` resolver
   * I wanted to extract this logic from the service itself so that we can keep it clean and not add extra response types to the service
* Add `TestSessionAndStudentResponseDTO` as new GraphQL response type
* Add GET_TEST_SESSION query


## Steps to test
1. Playground API call using test session id `6466dac26a580563b69514a2`

Successful call shows student data
<img width="770" alt="image" src="https://github.com/uwblueprint/jump-math/assets/69697180/2def9c75-64f9-44ef-82ef-e3ff8d217275">
<img width="772" alt="image" src="https://github.com/uwblueprint/jump-math/assets/69697180/27d8d66c-94f0-43aa-b3f8-b2251bacfac3">

Error if student doesn't exist (we should never have to run into this case tho given our UI)
<img width="768" alt="image" src="https://github.com/uwblueprint/jump-math/assets/69697180/22e65f64-f0e9-44c5-b80e-d21eda4add7d">



## What should reviewers focus on?
* Is this the best implementation in terms of performance and code cleanliness?

## Checklist
